drivers: ieee802154: cc13xx_cc26xx: use multi-protocol radio patch

cfriedt opened the pull request:
This change enables the multi-protocol RF patch to be used for the cc13xx_cc26xx IEEE 802.15.4 2.4 GHz PHY, which allows both the 2.4 GHz and Sub GHz PHY to be used simultaneously.

Eventually, BLE will also work simultaneously on 2.4 GHz (with arbitration).

Fixes #29883

galak reviewed (commented):
Can you have this point to the hal_ti repo and PR for testing.

Look here for details:

https://docs.zephyrproject.org/latest/guides/modules.html?highlight=modules#process-for-submitting-changes-to-existing-modules

cfriedt commented:
Relates to zephyrproject-rtos/hal_ti#22

cfriedt committed:
drivers: ieee802154: cc13xx_cc26xx: use multi-protocol radio patch

This change enables the multi-protocol rf patch to be used for
the cc13xx_cc26xx IEEE 802.15.4 2.4 GHz PHY, which allows both
the 2.4 GHz and Sub GHz PHY to be used simultaneously.

Eventually, BLE will also work simultaneously on 2.4 GHz (with
arbitration).

Fixes #29883

Signed-off-by: Christopher Friedt <chrisfriedt@gmail.com>

cfriedt commented:
Hahahaa... I just realized that this wasn't yet merged. Thank goodness github has a "restore branch" feature!😅
